### PR TITLE
extend repositories

### PR DESCRIPTION
**Pull request by tbr2**
We can now find authors + add new cheep / author

### CODE REVIEW

**Comment by chrolle69** [comment]
jeg får den her fejl når jeg prøver dotnet run:
Unhandled exception. Microsoft.Data.Sqlite.SqliteException (0x80004005): SQLite Error 1: 'no such table: Authors'.

får i andre ogs den? eller er der noget som jeg skal gøre før jeg kører dotnet run?

File: src/Chirp.Infrastructure/Repositories/AuthorRepository.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+            .FirstAsync();
+    }
+    public Task<AuthorDto> GetAuthorByEmail(string email) {
+        return _dbContext.Authors
+            .Where(author => author.Email == email)
+            .Select(author => new AuthorDto(author.Name, author.Email))
+            .FirstAsync();
+    }
+    public async Task CreateAuthor(AuthorDto author) {
+        await _dbContext.Authors.AddAsync(new Author() {Id = Guid.NewGuid(), Email = author.Email, Name = author.Name, Cheeps = new List<Cheep>()});
```
Comment (rafj58) [inline src/Chirp.Infrastructure/Repositories/AuthorRepository.cs:25]: I might be wrong but for the data to be persistent we need to save the changes to the database with the method _dbContext.SaveChanges().

File: src/Chirp.Infrastructure/Repositories/CheepRepository.cs [hunk 2]
```diff
@@ -4,16 +4,30 @@
...
-            .Where( cheep => cheep.Author.Name == author || author == null)
+    public Task<List<CheepDto>> GetCheeps(int page, string? author = null) {
+        return _dbContext.Cheeps
+            .Where(cheep => cheep.Author.Name == author || author == null)
             .OrderByDescending(cheep => cheep.TimeStamp)
-            .Skip( 32 * (page - 1))
+            .Skip(32 * (page - 1))
             .Take(32)
-            .Select( cheep => new CheepDto(cheep.Author.Name, cheep.Text, cheep.TimeStamp.ToString("MM/dd/yy H:mm:ss")))
+            .Select(cheep => new CheepDto(cheep.Author.Name, cheep.Text, cheep.TimeStamp.ToString("MM/dd/yy H:mm:ss")))
```
Comment (rafj58) [inline src/Chirp.Infrastructure/Repositories/CheepRepository.cs:18]: Why do we use the American way of writing dates? Is it a requirement?

File: src/Chirp.Infrastructure/Repositories/CheepRepository.cs [hunk 3]
```diff
@@ -4,16 +4,30 @@
...
+    public async Task AddCheep(string message, string authorName) {
+        // We have no way of knowing the correct email if author does not exist already
+        var author = _dbContext.Authors.Where(author => author.Name == authorName).FirstOrDefault(new Author() {
+            Id = Guid.NewGuid(),
+            Name = authorName,
+            Email = "thisisalegitemail@email.email",
+            Cheeps = new List<Cheep>()
+        });
+
+        await _dbContext.Cheeps.AddAsync(new Cheep() { Id = Guid.NewGuid(), AuthorId = author.Id, Author = author, Text = message });
```
Comment (rafj58) [inline src/Chirp.Infrastructure/Repositories/CheepRepository.cs:31]: As commented on other code that add an entry in the database. There should probably be a call to _context.SaveChanges().

**Review by rafj58 — APPROVED** [review]
Seems good

**Pull request merged**